### PR TITLE
Action cards

### PR DESCRIPTION
**Pull request by katebeavis**
Add `<ActionCard />` component that optionally renders a spinner when clicked

### CODE REVIEW

File: src/components/molecules/RadioGroupField/RadioGroupField.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ const RadioWrapper = styled.div`
 
 type RadioGroupFieldItem = {
   value: string;
-  label?: string | React.ReactNode;
```
Comment (moshie) [inline src/components/molecules/RadioGroupField/RadioGroupField.tsx:16]: [Question] How come `string` was removed?

Comment (katebeavis) [reply src/components/molecules/RadioGroupField/RadioGroupField.tsx:16]: This was some cleanup from a previous comment https://github.com/zopaUK/react-components/pull/564#discussion_r458651841

File: src/components/molecules/ActionCard/ActionCard.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  align-items: center;
+  justify-content: space-between;
+  flex-direction: row;
+`;
+
+const ActionCard: FC<ActionCardProps> = ({ children, id, icon = faChevronRight, loading, handleClick }) => {
+  const [selectedId, setSelectedId] = useState<string>('');
+  const handleOnClick = () => {
+    setSelectedId(id);
+    handleClick(id);
```
Comment (moshie) [inline src/components/molecules/ActionCard/ActionCard.tsx:46]: [Amendment] I would use the native onClick event handler here

```tsx
const handleOnClick = (event) => {
    setSelectedId(id);
    onClick(event, id);
}
```

File: src/components/molecules/ActionCard/ActionCard.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+`;
+
+const StyledActionCardContent = styled(Card.Content)`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  flex-direction: row;
+`;
+
+const ActionCard: FC<ActionCardProps> = ({ children, id, icon = faChevronRight, loading, handleClick }) => {
```
Comment (moshie) [inline src/components/molecules/ActionCard/ActionCard.tsx:42]: [Amendment] change `handleClick` to `onClick`

```tsx
const ActionCard: FC<ActionCardProps> = ({ children, id, icon = faChevronRight, loading, onClick }) => {
```

**Review by moshie — CHANGES_REQUESTED** [review]
Looks good couple of suggestions

File: src/components/molecules/RadioGroupField/RadioGroupField.tsx (hunk 1 above)
Comment (poteirard) [reply src/components/molecules/RadioGroupField/RadioGroupField.tsx:16]: `React.ReactNode` already includes `string`

File: src/components/molecules/ActionCard/ActionCard.test.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      </ActionCard>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+    const results = await axe(container.innerHTML);
+    expect(results).toHaveNoViolations();
+  });
+
+  it('should render the action card when loading is true and the card is clicked', () => {
+    const { container } = render(
+      <ActionCard id="test-card" loading={true} handleClick={() => console.log('test')}>
```
Comment (admmasters) [inline src/components/molecules/ActionCard/ActionCard.test.tsx:20]: Can use the shortcut for loading as
```
<ActionCard loading />
```

File: src/components/molecules/ActionCard/ActionCard.test.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        Action Card Content
+      </ActionCard>,
+    );
+    fireEvent.click(container);
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it('should render the action card when loading is true and the card is clicked', () => {
+    const { container } = render(
+      <ActionCard id="test-card" loading={false} handleClick={() => console.log('test')}>
```
Comment (admmasters) [inline src/components/molecules/ActionCard/ActionCard.test.tsx:30]: presumably `loading={false}` is implicit in not passing a loading property

File: src/components/molecules/ActionCard/ActionCard.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import React, { FC, useState } from 'react';
+import Card from '../../organisms/Card';
+import { colors } from '../../../constants';
+import Spinner from '../../atoms/Spinner/Spinner';
+import Icon from '../../atoms/Icon/Icon';
+import { faChevronRight, IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import styled from 'styled-components';
+
+interface ActionCardProps {
+  children: any;
```
Comment (admmasters) [inline src/components/molecules/ActionCard/ActionCard.tsx:10]: React.ReactNode

File: src/components/molecules/ActionCard/ActionCard.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+const ActionCard: FC<ActionCardProps> = ({ children, id, icon = faChevronRight, loading, onClick }) => {
+  const [selectedId, setSelectedId] = useState<string>('');
+  const handleOnClick = (event: React.MouseEvent | React.KeyboardEvent) => {
+    setSelectedId(id);
+    onClick(event, id);
+  };
+  return (
+    <StyledActionCard onClick={handleOnClick} data-automation="ZA.action-card">
+      <StyledActionCardContent>
+        <div>{children}</div>
```
Comment (admmasters) [inline src/components/molecules/ActionCard/ActionCard.tsx:51]: Does it need to be a div or could it be a fragment?

**Pull request closed without merging**